File: Parksujin/[PGS] 로또의 최고순위와 최저순위.py
# lottos가 win_nums에 있다면 rank +1
# 최고순위 = lottos의 0 개수를 센 후 모두 rank에 더해줌
# 최저순위 = win_nums와 다른 수들은 겹치지 않는다.
# 1개번호 일치 & 모두 불일치 => 6위
def solution(lottos, win_nums):
    answer = []
    rank = 0
    zero = 0

    for i in lottos:
        if i != 0:
            if i in win_nums:
                rank += 1
        else:
            zero += 1

    #최고순위
    max = 7 - (rank + zero)

   # 최저순위
    min = 7 - rank

    if max > 6 : max = 6
    if min > 6 : min = 6

    return [max, min]

File: Kanghyun/[PGS] 로또의 최고 순위와 최저 순위.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int Rank(int n) {
    int rank = 6;
    if (n == 6)
        rank = 1;
    else if (n == 5)
        rank = 2;
    else if (n == 4)
        rank = 3;
    else if (n == 3)
        rank = 4;
    else if (n == 2)
        rank = 5;
    return rank;
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int same_numbers = 0;
    int unidentified = 0;

    for (int i = 0; i < 6; i++)
        if (*find(win_nums.begin(), win_nums.end(), lottos[i]) == lottos[i])
            same_numbers += 1;

    for (int i = 0; i < 6; i++)
        if (lottos[i] == 0)
            unidentified++;
    int max = same_numbers + unidentified;
    
    answer.push_back(Rank(max));
    answer.push_back(Rank(same_numbers));

    return answer;
}